### ISSUE TITLE
Home assitant cannot control sonoff devices

### ISSUE THREAD

**Issue by canvacanva** [AlexxIT/SonoffLAN#1323]
I got 3 switches, 4 fans and a relay by sonoff ( mini R2/ ifan rw and r3), from Yesterday i cannot control none from HA. I already update HA and sonoff plugin with no success.
Control from Alexa or ewelink is working.
HA 2024.1.3
Sonoff 3.5.4

**Comment by vulpasin** [comment]
Invalid config
The following integrations and platforms could not be set up: sonoff

Same problems after HA 2024.1.3 update.

**Comment by xmicky** [comment]
Roll back to HA 2023.12.3
Use advanced ssh & web terminal 
ha core update —version 2023.12.3
The last one that works with the current sonofflan 
See the other posts about this

**Comment by DaSpors** [comment]
> Roll back to HA 2023.12.3
> Use advanced ssh & web terminal
> ha core update —version 2023.12.3
> The last one that works with the current sonofflan
> See the other posts about this

Same over here

**Comment by beta-j** [comment]
2023.12.4 seems to be working fine for me too.

It's 2024.1.x that seems to be misbehaving with Sonoff LAN :(

**Comment by patrickgmail** [comment]
Got some problems too with 2024.1.5 buyt it's strange.
Problems are only with 3 T1  (2c and 3c) with firmware 3.7.6 while others T1 with same firmware are working well.
No problem with basic, mini, TH16...
It seems that Ewelink app have problems too with those problematics switches.

**Comment by xmicky** [comment]
It all works now for me with the latest  ha core 2024.1.5 and Sonoff lan 3.5.4

**Comment by patrickgmail** [comment]
> Got some problems too with 2024.1.5 buyt it's strange. Problems are only with 3 T1 (2c and 3c) with firmware 3.7.6 while others T1 with same firmware are working well. No problem with basic, mini, TH16... It seems that Ewelink app have problems too with those problematics switches.

I found what was wrong on mi side: many users and after setup a new instance of HA with 2024.1.5, I had forgotten to declare one. So, in my case:
**All runs well with 2024.1.5 (odcker container) and Sonofflan 3.5.4**

**Comment by DaSpors** [comment]
> It all works now for me with the latest ha core 2024.1.5 and Sonoff lan 3.5.4

Can confirm that too!

**Comment by jahnroux** [comment]
I tried to upgrade to HA core 2024.1.5 and SonoffLan 3.5.4 but some things were still not working well, for example I had no control over my Sonoff B02 bulb. Some of my other switches were not reporting correct state. I would turn on a switch and it would turn on, but then in HA it reverts back to showing off, meanwhile the switch is still on and ewelink shows switch as on. Rolled back HA core to 2023.12.3 and immediately everything worked again.

Ran the following in terminal: ha core update —version 2023.12.3
